node/src/components/consensus/era_supervisor.rs

```diff
@@ -402,7 +396,7 @@ where
             trace!("executed block in old era {}", block_header.era_id().0);
             return effects;
         }
-        if block_header.switch_block() {
+        if block_header.era_end().is_some() {
```

Comment:
Not sure if I prefer `switch_block()` over this. Seemed to me like it's our [ubiquitous language](https://www.martinfowler.com/bliki/UbiquitousLanguage.html).

Reply:
Yes, I wasn't sure about it either. It just feels weird to call the type `SwitchBlock` even though it's only a small part of a block. `SwitchBlockInfo`? (Although using `Info` in any identifier always feels like a bad excuse…)

In the same vein, I use `equivocator`, `faulty` and `slashing` with overlapping meanings sometimes… :confused:

Reply:
I would just add `is_switch_block()` method on the block header that delegates to this.